File: backend/TasksAggregate/Task.cs
using backend.TasksAggregate.ViewModels;

namespace backend.TasksAggregate;

public class Task
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }

    public Task(string title, string description)
    {
        Title = title;
        Description = description;
    }

    public Task(TaskEditViewModel taskEditViewModel)
    {
        Title = taskEditViewModel.Title;
        Description = taskEditViewModel.Description;
    }
}

File: frontend-demo/src/taskAggregate/domain/Task.ts
export class Task {
    public id: number;
    public title: string;
    public description: string;
    
    constructor(id: number, title: string, description: string) {
        this.id = id;
        this.title = title;
        this.description = description;
    }
    
    public EmptyFields() {
        return new Task(0, "", "");
    }
    
    public SetTitle(title: string) {
        return new Task(this.id, title, this.description);
    }
    
    public SetDescription(description: string) {
        return new Task(this.id, this.title, description);
    }
}